idea-plugin/src/main/kotlin/arrow/meta/ide/plugins/proofs/ProofsIdePlugin.kt

```diff
@@ -51,11 +52,54 @@ val IdeMetaPlugin.typeProofsIde: Plugin
                 }
               }
             }
-
-//            holder.createErrorAnnotation(f, "This is a call element")?.let { error ->
-//             // error.registerUniversalFix(AddModifierFix(f, KtTokens.SUSPEND_KEYWORD), f.identifyingElement?.textRange, null)
-//            }
+//          holder.createErrorAnnotation(f, "This is a call element")?.let { error ->
+//            // error.registerUniversalFix(AddModifierFix(f, KtTokens.SUSPEND_KEYWORD), f.identifyingElement?.textRange, null)
+//          }
           }
+        }),
+      addAnnotator(
+        annotator = Annotator { element: PsiElement, holder -> // in some situations there are 2 or more error annotations
+          element.safeAs<KtCallElement>()?.let { ktCall ->
+            val ctx = ktCall.analyze(bodyResolveMode = BodyResolveMode.FULL)
+            val calls = ctx.getSliceContents(BindingContext.RESOLVED_CALL)
+            calls.filter { it.value != null }
+              .all { (call, resolvedCall) ->
+                if (call.callElement != element) false
+                else {
+                  val isSubtypeOf = baseLineTypeChecker.isSubtypeOf(resolvedCall.getReturnType(), resolvedCall.candidateDescriptor.returnType!!)
+                  val module = resolvedCall.resultingDescriptor.module
+                  val compilerContext = CompilerContext(project = call.callElement.project, eval = {
+                    KotlinJsr223StandardScriptEngineFactory4Idea().scriptEngine.eval(it)
+                  })
+                  compilerContext.module = module
+                  // val proofs = compilerContext.coerceProofs(ktCall.returnType!!, ktCall.calleeExpression?.getType(ctx)!!)
+                  val proofs = compilerContext.coerceProofs(resolvedCall.getReturnType(), resolvedCall.candidateDescriptor.returnType!!)
+                  isSubtypeOf && proofs.isNotEmpty()
+                }
+              }
+          }
+        }),
+      addIntention(
```

Comment:
There is no need to use `addIntention`, this is just duplicating boilerplate.
- Create an Annotation within the `Annotator` for your desired Element
```
holder.createInfoAnnotation(call, "Type proof coerced call-site")?.let { info ->
    info.registerUniversalFix(<MyInspection/ or MyIntention>)
```
- define what the Intention should do, here you can use functions from `InspectionSyntax` or `IntentionSyntax`

Reply:
Mmm for this specific use-case.. we have decided there is no need to highlight anything.. until you are with the cursor over it and only then provide the option to do the _fix_.. meaning make it implicit / explicit.. thats why we wanna use the `addIntention`
Still work in progress tho

Reply:
That is what an InfoAnnotation does

Reply:
But whatever works better for this use-case :)